contentcuration/contentcuration/frontend/shared/vuex/file/utils.js

```diff
@@ -44,6 +51,165 @@ export function getHash(file) {
+                ).then(data => {
+                  file.folders = data;
+                });
+              }
+              const metadataNodes = orgNode.getElementsByTagName('metadata');
+              if (metadataNodes && metadataNodes.length != 0) {
+                Object.values(metadataNodes).forEach(nodeValue => {
+                  file[`${nodeValue[0].nodeName}`] = nodeValue[0].textContent.replace(
+                    / {2}|\r\n|\n|\r/gm,
+                    ''
+                  );
+                });
+              }
+              org.files.push(file);
+              const manifestPath =
+                file.resourceHref.slice(0, file.resourceHref.lastIndexOf('/') + 1) +
+                'imsmanifest.xml';
+              const subManifestContent = zip.files[manifestPath];
+              if (subManifestContent && !procssedFiles.includes(manifestPath)) {
+                procssedFiles.push(manifestPath);
+                const subManifestFile = await Promise.resolve(subManifestContent.async('text'));
+                const parser = new DOMParser();
+                const xmlDoc = parser.parseFromString(subManifestFile, 'application/xml');
+                const subManifestData = await getFolderMetadata(
+                  xmlDoc.getElementsByTagName('organizations'),
+                  xmlDoc,
+                  zip,
+                  procssedFiles
+                );
+                if (subManifestData.title) {
+                  org.title = subManifestData[0].title;
+                }
+                subManifestData[0].files.map(file => {
+                  org.files.push(file);
+                });
+              }
+            })
+          );
+        }
+        folders.push(org);
+        return org;
+      })
+    );
+    return folders;
+  }
+}
+
+async function getManifestMetadata(manifestFile, zip, procssedFiles) {
+  const metadata = {};
+  const parser = new DOMParser();
+  const xmlDoc = parser.parseFromString(manifestFile, 'application/xml');
+  const data = xmlDoc.getElementsByTagName('organizations');
+  return await getFolderMetadata(data, xmlDoc, zip, procssedFiles).then(async data => {
+    if (data) {
+      metadata.folders = data;
+    }
+    const metadataFile = zip.file('imsmetadata.xml');
+    if (metadataFile) {
+      procssedFiles.push('imsmetadata.xml');
+      const content = await Promise.resolve(metadataFile.async('text'));
+      const xmlDoc = parser.parseFromString(content, 'application/xml');
+      if (xmlDoc.getElementsByTagName('lomes:title').length) {
+        metadata.title = xmlDoc
+          .getElementsByTagName('lomes:title')[0]
+          .children[0].textContent.replace(/ {2}|\r\n|\n|\r/gm, '');
+      }
+      if (
+        xmlDoc.getElementsByTagName('lomes:idiom').length &&
+        xmlDoc
+          .getElementsByTagName('lomes:idiom')[0]
+          .textContent.replace(/ {2}|\r\n|\n|\r/gm, '') !== 'und'
+      ) {
+        metadata.language = xmlDoc
```

Comment:
For language extraction let's replicate the validation we are doing in H5P to check this is a supported language code.

Reply:
Can also add some more tests for unhappy paths!